compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/GPUTileSwizzleUtils.cpp

```diff
@@ -144,38 +144,19 @@ TileSwizzle getIntrinsicSwizzle(IREE::GPU::MMAIntrinsic intrinsic,
   return swizzle;
 }
 
-// Returns the index of the dimension whose flattened size (flattening inner
-// dimensions into it) matches the given `targetSize`. This is used to compute
-// interleaving indices.
-//
-// Example:
-//    Input shape = [16, 8, 4, 4]
-//    Input targetSize = 16
-// -> Return 2, because the tail of the shape starting at index 2 is [4, 4],
-//    whose product equals targetSize.
-static int64_t
-getDimIdxForTargetSize(const TileSwizzle::ExpandShapeDimVectorType &shape,
-                       int64_t targetSize) {
-  int interleaveAt = 0;
-  int size = 1;
-  for (interleaveAt = shape.size() - 1; interleaveAt >= 0; --interleaveAt) {
-    assert(size <= targetSize);
-    assert((targetSize % size) == 0);
-    if (size == targetSize) {
+static int64_t getInnermostCrossThreadDimIdx(
+    const TileSwizzle::ExpandShapeDimVectorType &shape) {
```

Comment:
We are missing the logic when there are no CrossThread kind of tiles. If it is very wrong, we can add an assertion at the end. If not (i.e., expected input), we should make the return type `std::optiona<int64_t>` and ask the callers to handle the failure.

Reply:
I changed this function into `getInnermostNonInternalDimIdx` and now it's clear that we can just assert that not all of the dimensions are internal.